src/NuGet.Services.AzureSearch/BaseDocumentBuilder.cs

```diff
@@ -175,16 +179,21 @@ public BaseDocumentBuilder(IOptionsSnapshot<AzureSearchJobConfiguration> options
 
             if (leaf.IconFile != null)
             {
-                var provider = new FlatContainerPackagePathProvider(_options.Value.FlatContainerContainerName);
-                var iconPath = provider.GetIconPath(document.PackageId, normalizedVersion);
-                document.IconUrl = new Uri(_options.Value.ParseFlatContainerBaseUrl(), iconPath).AbsoluteUri;
+                SetIconUrlFromFlatContainer(document);
             }
             else
             {
                 document.IconUrl = leaf.IconUrl;
             }
         }
 
+        private void SetIconUrlFromFlatContainer(IBaseMetadataDocument document)
+        {
+            var provider = new FlatContainerPackagePathProvider(_options.Value.FlatContainerContainerName);
```

Comment:
shouldn't this be initialized in the constructor?

Reply:
Not sure--it was not previously, and I didn't want to change the behavior. Potentially, the `_options.Value.FlatContainerContainerName` could change and then require this to be reinitialized, but I don't know if that's possible.

@joelverhagen @loic-sharma @shishirx34 - is that a possibility?

Reply:
You could since configs never change at runtime. But IIRC there are some option interfaces in Microsoft.Extensions.Options/Configuration that can give different values over time so I personally try to "new up" classes like these in method bodies we we can have fewer "caches"

Ideally `FlatContainerPackagePathProvider` would itself accept a config option in ctor and leave this kind of decision as its own implementation detail.

Reply:
Given that, I'd rather leave this as-is.